src/effects/effectslot.cpp

```diff
@@ -175,40 +159,23 @@ EffectManifestPointer EffectSlot::getManifest() const {
     return m_pManifest;
 }
 
-EffectKnobParameterSlotPointer EffectSlot::addEffectKnobParameterSlot() {
-    auto pParameterSlot = EffectKnobParameterSlotPointer(
-            new EffectKnobParameterSlot(m_group, m_knobParameterSlots.size()));
-    m_knobParameterSlots.append(pParameterSlot);
-    m_pControlNumParameterSlots->forceSet(
-            m_pControlNumParameterSlots->get() + 1);
-    return pParameterSlot;
-}
-
-EffectButtonParameterSlotPointer EffectSlot::addEffectButtonParameterSlot() {
-    auto pParameterSlot = EffectButtonParameterSlotPointer(
-            new EffectButtonParameterSlot(m_group, m_buttonParameterSlots.size()));
-    m_buttonParameterSlots.append(pParameterSlot);
-    m_pControlNumButtonParameterSlots->forceSet(
-            m_pControlNumButtonParameterSlots->get() + 1);
-    return pParameterSlot;
-}
-
-unsigned int EffectSlot::numKnobParameters() const {
-    unsigned int num = 0;
-    for (auto const& pParameter : m_parameters) {
-        if (pParameter->manifest()->controlHint() !=
-                EffectManifestParameter::ControlHint::TOGGLE_STEPPING) {
-            ++num;
-        }
+void EffectSlot::addEffectParameterSlot(EffectParameterType parameterType) {
+    EffectParameterSlotBasePointer pParameterSlot = nullptr;
+    if (parameterType == EffectParameterType::Knob) {
+        pParameterSlot = new EffectKnobParameterSlot(m_group, m_iNumParameterSlots[EffectParameterType::Knob]);
+        ++m_iNumParameterSlots[EffectParameterType::Knob];
+    } else if (parameterType == EffectParameterType::Button) {
+        pParameterSlot = new EffectButtonParameterSlot(m_group, m_iNumParameterSlots[EffectParameterType::Button]);
+        ++m_iNumParameterSlots[EffectParameterType::Button];
     }
-    return num;
+    m_parameterSlots.append(pParameterSlot);
 }
 
-unsigned int EffectSlot::numButtonParameters() const {
+unsigned int EffectSlot::numParameters(EffectParameterType parameterType) const {
```

Comment:
This function is unused. Other classes should not have access to this implementation detail anyway. Please remove this function.

Reply:
It is being used in DlgPrefEq class. I think we can implement a `nextParameter(EffectParameterType parameterType, int index)` method in EffectSlot and remove this function. What do you think?